### ISSUE TITLE
Anchor Tags In Gallery Images Are Covered by the Caption, Rendering Them Unclickable

### ISSUE THREAD

**Issue by Steve-MP** [WordPress/gutenberg#14304]
**Describe the bug**
The Gutenberg Gallery block renders each image with the caption overlaid over the image. This is done by absolute positioning of the figcaption element. Because the caption is not rendered inside the anchor tag which links each image to its source file, the area covered by the caption is not clickable. This is unexpected behaviour from a user perspective and provides a bad experience. Users expect the whole image/caption block to be clickable. Because the figcaption is rendered with a transparent backdrop, it is not clear to the user why the clickable area of the image seems to arbitrarily end half way down the image.

**To Reproduce**
1. Create a post
2. Insert a Gutenberg Gallery block
3. Add an image with a long caption and select Link To: Media File from the block menu to the right of the editor interface
4. Save the post
5. Visit the post page
6. Attempt to click the image at various places - note that the clickable area ends at an arbitrary point.

**Expected behavior**
The whole image should be clickable, even if a caption is present. If there is a caption, the caption should be included in the anchor tag.

**Screenshots**
![clicky](https://user-images.githubusercontent.com/13947767/53947902-0189fa00-40c7-11e9-9b02-7b6b00d7e445.png)

**Desktop (please complete the following information):**
 - OS: OSX Mojave
- Google Chrome for Mac v. 72.0.3626.121

**Additional context**
- The version in Wordpress core 5.1

**Comment by youknowriad** [comment]
Should we wrap the caption inside the link cc @jasmussen @mapk ?

**Comment by jasmussen** [comment]
Can we add `pointer-events: none;` to the caption? I don't know if we can add the caption inside the link, as captions can be multiline (i.e. they are wrapped in P) and a p nested inside an a tag is invalid HTML.

**Comment by Steve-MP** [comment]
> Can we add `pointer-events: none;` to the caption? I don't know if we can add the caption inside the link, as captions can be multiline (i.e. they are wrapped in P) and a p nested inside an a tag is invalid HTML.

I think that in HTML5 you can wrap an anchor around pretty much anything apart from other interactive elements. Wrapping block level elements like p inside an a is fine.

**Comment by mapk** [comment]
> I think that in HTML5 you can wrap an anchor around pretty much anything apart from other interactive elements.

Let's move in this direction. Are there any a11y issues that we should anticipate regarding this change?

**Comment by nosolosw** [comment]
Related https://github.com/WordPress/gutenberg/issues/13164

**Comment by ashwin-pc** [comment]
Hey! I saw the `Needs Dev` label. I'd love to take this issue up.

**Comment by afercia** [comment]
Before looking at the PR, I'd like to point out this problem already existed in the pre-Gutenberg era. However, it actually depended on the _theme_. 

For example, some of the bundled themes render the classic gallery captions in a way that doesn't overlay the image, see Twenty Seventeen:

![gallery twenty seventeen](https://user-images.githubusercontent.com/1682452/55879794-b66c7680-5b9f-11e9-86e5-db8dae199b48.jpg)

Instead, other bundled themes do overlay the caption onto the image e.g. Twenty Fourteen (same for Twenty Thirteen):

![gallery twenty fourteen](https://user-images.githubusercontent.com/1682452/55879856-d69c3580-5b9f-11e9-9786-2dd6c124cff2.jpg)

So this problem is not new to Gutenberg and it's actually a design issue.

Worth also noting the classic gallery in Twenty Nineteen doesn't overlay the captions:

![classic gallery twenty nineteen](https://user-images.githubusercontent.com/1682452/55880000-23800c00-5ba0-11e9-8e72-8bbacae2bcce.jpg)

While the Gutenberg gallery does overlay captions:

![gutenberg gallery twenty nineteen](https://user-images.githubusercontent.com/1682452/55880037-3397eb80-5ba0-11e9-8945-cf5c805efda2.jpg)

So on one hand, Gutenberg makes an opinionated choice and forces the captions to overlay the image. On the other hand, it doesn't provide a solution to the clickable area issue.

I'm not sure forcing this rendering is the best option. To me, this should be handled by the theme and should be a theme author's choice.

Will post accessibility related considerations [on the PR](https://github.com/WordPress/gutenberg/pull/14888).

**Comment by mapk** [comment]
@ashwin-pc thanks for jumping into this! 👍

**Comment by mapk** [comment]
> I'm not sure forcing this rendering is the best option. To me, this should be handled by the theme and should be a theme author's choice.

Good point @afercia. I wonder if the two different styles can be block style variations as [mentioned in this comment](https://github.com/WordPress/gutenberg/issues/11436#issuecomment-481297311).

**Comment by afercia** [comment]
@maok yep commented also on #11436. I'd totally second your point: style variations could really make sense here.

**Comment by gziolo** [comment]
@ashwin-pc did some explorations in #11436 which was concluded with the following recommendation:

> > does the caption need to stay visible when the image is hovered? -> consider to hide it, or truncate it, or reduce its height on hover (or any other "smart" behavior)
> Can we try this? Hide the caption on hover if the caption covers the image.

It was also noted by @mapk that this isn't a perfect solution but it's fine to fix the bug:

> I'm willing to take this tradeoff because I believe clicking an image to view it at a larger scale is the prefered action here. This PR can work toward this, and in the future we can create a PR that adds another style variation where the captions are below the images.

**Comment by glendaviesnz** [comment]
> Can we add `pointer-events: none;` to the caption? I don't know if we can add the caption inside the link, as captions can be multiline (i.e. they are wrapped in P) and a p nested inside an a tag is invalid HTML

This approach seems to work with the new innerBlocks Gallery @jasmussen  - [PR here](https://github.com/WordPress/gutenberg/pull/40713).

**Comment by ndiego** [comment]
@glendaviesnz there was some conversation about this issue in the Editor Weekly Bug Scrub, here is the [direct link](https://wordpress.slack.com/archives/C02QB2JS7/p1656427622691619) to the conversation. The consensus after reading through all the comments on this issue and the associated closed PRs, is that this is expected behavior and the issue should be closed. While I personally agree that WordPress Core likely should not have added the caption overlay to begin with since captions can also have links, this issue can be "resolved" with an additional block style or through specific theme styles. The block style approach would be an enhancement PR in my mind. 

So, what are your thoughts on closing this issue?

**Comment by glendaviesnz** [comment]
@ndiego I agree with that conclusion.
